Feature/auth integration

LarsSonke opened the pull request:
# Add Broker Service with Status Endpoint

Implementeert broker service in de `broker/` directory met een `/api/v1/status` endpoint.

## Features
- ✅ Status endpoint toont system health en timestamp
- ✅ Optionele JWT authenticatie - toont user info bij geldige token
- ✅ Test script (`test-broker.ps1`) toegevoegd

## Gebruik
```bash
cd broker
go run main.go

LarsSonke committed:
Add broker service with status endpoint and optional JWT authentication

LarsSonke committed:
Add test script and environment configuration for broker service

LarsSonke committed:
Fix unused import and add gitignore for broker service

Jimmaphy commented on broker/.gitignore:
Please add this information to the .gitignore file in the root of the project.

Jimmaphy commented on broker/README.md:
Graag verplaatsen naar een, in lowercase, readme.md bestand in de root van het project.

Jimmaphy commented on test-broker.ps1:
We have mixed operating system within the group. This script is only executable on Windows. I can't even get Julien to install Microsoft Office, let alone PowerShell. Try and do it with go unit testing. Possible solutions:

- Create a Go file that uses the testing package and os/exec to achieve the same goal.
- Use Testinfra in Python to test the compiled binary.
- Use PowerShell but wrap it in a Docker container so it can easily be integrated into a CI/CD pipeline if we ever want to.

Jimmaphy reviewed (changes requested):
I've left some half Dutch half English comments with my toughts :D

LarsSonke committed:
feat: add plugin registration with JWT auth and persistence

- Add plugin registration endpoints (POST /api/v1/route, GET /api/v1/routes)
- Implement RequireAuth middleware for JWT-based authentication
- Add Plugin model and in-memory registry with atomic file persistence
- Make persistence path configurable via PLUGINS_PERSIST_PATH env var
- Add APP_VERSION config and move version out of hardcoded status handler
- Add warning logs for failed JWT_EXPIRY parsing and PublicKeyPEM errors
- Add cross-platform Go unit tests for status and plugin endpoints
- Consolidate documentation into root readme.md (lowercase)
- Remove broker.exe binary artifact

Tests cover:
- Status endpoint with/without auth
- Plugin registration auth requirements
- Successful registration with persistence
- Duplicate slug conflict handling

LarsSonke committed:
feat: add plugin update/delete endpoints and route conflict validation

- Add Update and Delete methods to plugin registry with persistence
- Implement PUT /api/v1/route/:slug for plugin updates (requires auth)
- Implement DELETE /api/v1/route/:slug for plugin deletion (requires auth)
- Add path normalization (leading slash, no trailing slash)
- Add route conflict validation to prevent duplicate base-api-routes
- Validate plugins don't conflict with reserved broker routes
- Add comprehensive tests for update, delete, and conflict scenarios

All tests passing (ok broker/internal/handlers 1.663s)
